ipld/write.go

```diff
@@ -55,3 +48,19 @@ func ExtractODSShares(eds *rsmt2d.ExtendedDataSquare) [][]byte {
 	}
 	return origShares
 }
+
+// batchSize calculates the amount of nodes that are generated from block of 'squareSizes'
+// to be batched in one write.
+func batchSize(squareSize int) int {
+	// (squareSize*squareSize) - all the shares
+	// (squareSize*2-1) - amount of nodes in a generated binary tree
+	// *squareSize*2 - multiplier for the amount of trees, both over rows and cols
+	//
+	// Note that our IPLD tree looks like:
+	// ---X
+	// -X---X
+	// X-X-X-X
+	// X-X-X-X
+	// So we count leaves two times here for a reason. https://github.com/celestiaorg/celestia-node/issues/183
+	return squareSize*squareSize + (squareSize*2-1)*squareSize*2
```

Comment:
OK, so this is really weird but the numbers of keys readable via `bs.AllKeysChan(ctx)` as per test in https://github.com/celestiaorg/celestia-node/pull/471
always matches this formula instead: 
```suggestion
	return (squareSize*2-1)*squareSize*2 - (squareSize * squareSize)
```

Any idea why we can remove the a CID per leaf/share?

Reply:
No idea right now. Will thing with clear mind tmrw

Reply:
Ah, I think i understand: the shares / leaves yield the same CIDs independent of if we compute row or column trees. But in `(squareSize*2-1)*squareSize*2` we compute all  hashes for leaves for rows and columns too (where the leaves yield the same keys/CIDs). hence we have to subtract these.
I guess the CIDs for the last two layers are the same as well? here:
// X-X-X-X
// X-X-X-X

Reply:
Ohhhhh, yesss

Reply:
This explains it

Reply:
That additional layer is always the same for rows and col.